scapy/contrib/automotive/someip.py

```diff
@@ -0,0 +1,151 @@
+#! /usr/bin/env python
+
+# MIT License
+
+# Copyright (c) 2018 Jose Amores
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Sebastian Baar <sebastian.baar@gmx.de>
+# This program is published under a GPLv2 license
+
+from scapy.layers.inet import TCP, UDP
+from scapy.all import struct, bind_layers
+from scapy.packet import Packet
+from scapy.fields import ShortField, BitEnumField, ConditionalField, \
+    BitField, PacketField, IntField, ByteField, ByteEnumField
+
+
+class _SOMEIP_MessageId(Packet):
+    """ MessageId subpacket."""
+    name = "MessageId"
+    fields_desc = [
+        ShortField("srv_id", 0),
+        BitEnumField("sub_id", 0, 1, {0: "METHOD_ID", 1: "EVENT_ID"}),
+        ConditionalField(BitField("method_id", 0, 15),
+                         lambda pkt: pkt.sub_id == 0),
+        ConditionalField(BitField("event_id", 0, 15),
+                         lambda pkt: pkt.sub_id == 1)
+    ]
+
+    def extract_padding(self, p):
```

Comment:
Please use `s` instead of `p`.

Reply:
Thanks a lot for accepting my remarks. This one is still relevant (that's because the argument name is `s` in the parent class; I know it is not applied everywhere in Scapy but we try to make new code better).

Reply:
It is implemented in the commit, I just forget to write my "done" under your remark, sorry about that.

Reply:
Can you double check? I have one `extract_padding` with the good argument name, and one with the other.

Reply:
I see it now, I overlooked it. Sorry my bad. It is fixed in the new commit and thanks again for your remarks.